climpred/relative_entropy.py

```diff
@@ -0,0 +1,343 @@
+    """
+    Add a `len(dim_label)` dimension `dim` to uninitialized control with time_dim by bootstrapping.
+
+    """
+    c_start = 0
+    c_end = control[time_dim].size
+    time = np.arange(1, 1 + lead_years)
+
+    def isel_years(control, year_s, m=None, length=lead_years):
+        new = control.isel({time_dim: slice(year_s, year_s + length - 0)})
+        new = new.rename({time_dim: 'time'})
+        if isinstance(new, xr.DataArray):
+            new['time'] = time
+        elif isinstance(new, xr.Dataset):
+            new = new.assign(time=time)
+        return new
+
+    def create_pseudo_members(control):
+        # print(c_start,c_end,lead_years,dim_label)
+        startlist = np.random.randint(
+            c_start, c_end - lead_years - 1, len(dim_label))
+        return xr.concat([isel_years(control, start)
+                          for start in startlist], dim)
+
+    control_uninitialized = create_pseudo_members(
+        control).assign({dim: dim_label})
+    return control_uninitialized
+
+
+def compute_relative_entropy(initialized, control,
+                             anomaly_data=False, neofs=None, curv=True,
+                             ntime=None, detrend_by_control_unitialized=True,
+                             nmember_control=10):
+    """
+    Compute relative entropy.
+
+    Calculates EOFs from anomalies. Projects fields on EOFs to receive
+    pseudo-Principle Components per initialization and lead year. Calculate
+    relative entropy based on _relative_entropy_formula.
+
+    Args:
+        initialized (xr.Dataset): anomaly ensemble data with dimensions
+                                    initialization, member, time and spatial
+                                    [lon (x), lat(y)]. DPLE or PM_ds
+        control (xr.Dataset): anomaly control distribution with
+                                              non-spatial dimensions:
+                                              spatial [lon (x), lat(y)].
+                                              - LENS: member, time
+                                              - PM_control: initialization
+        anomaly_data (bool): Input data is anomaly alread. Default: False.
+        neofs (int): number of EOFs to use. Default: initialized.member.size.
+        curv (bool): if curvilinear grids are provided disables EOF weights.
+        ntime (int): number of timesteps calculated.
+        detrend_by_control_uninitialized (bool): Default: True
+        nmember_control (int): number of members created from bootstrapping from control
+
+    Returns:
+        rel_ent (pd.DataFrame): relative entropy
+
+    """
+    # Defaults
+    if neofs is None:
+        neofs = initialized.member.size
+    if ntime is None:
+        ntime = initialized.time.size
+
+    # case if you only submit control with dim time, PM case
+    if ('ensemble' not in control.dims) and ('member' not in control.dims):
```

Comment:
What is the purpose of `ensemble` here? Should this be `initialization`? I imagine the PM case should just be `if 'member' not in control.dims` and then else will capture DPLE style cases.

Reply:
should have been `initialisation`